[DOCS-12318] Update OP Scaling Best Practices (#33694)
DataDog/documentation: content/en/observability_pipelines/scaling_and_performance/best_practices_for_scaling_observability_pipelines.md

## Overview

Deploy the Observability Pipelines Worker into your infrastructure, like you would any other service, to intercept, manipulate, and forward data to your destinations. Each Observability Pipelines Worker instance is designed to operate independently, allowing you to scale your architecture using simple load balancing.

This guide walks you through the recommended aggregator pattern for new Observability Pipelines Worker users, specifically:

- [Architecture models and approaches](#architecture)
- [Optimizing the instance](#optimize-the-instance) so you can horizontally scale the Observability Pipelines Worker aggregator.
- Starting points to estimate your resource capacity for [capacity planning and scaling](#capacity-planning-and-scaling) the Observability Pipelines Worker.

## Architecture

> This section goes over:

Reviewer: ```suggestion
This section covers:
```

- Architecture models:
	- [VM-based model](#vm-based-architecture)

> 	- [Kubernetes- based model](#kubernetes-based-architecture)

Reviewer: ```suggestion
	- [Kubernetes-based model](#kubernetes-based-architecture)
```

- Your organization uses Kubernetes heavily and is proficient with it

Choose the VM-based architecture if your organization is more VM centric and not proficient with Kubernetes.

Choosing between the two models comes down to what your organization is best equipped to do from an infrastructure perspective. Each model offers the ability to automatically scale based on CPU utilization, which is generally the primary constraint for Observability Pipelines. See [Optimize the instance][6] for more information.

### Centralized vs decentralized approach

Datadog recommends the decentralized approach of deploying the Workers as close to the data source as possible. This means placing Workers within each location where the data originates, such as the region, cluster, or datacenter. The decentralized model is better for environments with large volumes of data because it:

- Minimizes cross-region or cross-datacenter network transit
- Avoids potential performance issues related to inter-region or inter-account data transfer
- Helps reduce data transfer costs by keeping processing local to the data sources

> The centralized deployment model involves deploying Workers in a single location and sending data from multiple origins, such as regions, clusters, datacenters, and so on, to that central location. This approach may be suitable when handling lower volumes of data or when network peering between environments is already in place. However, centralized deployments can lead to increased data transfer fees, especially when sending high volumes of data across regions or accounts.

Reviewer: ```suggestion
A centralized deployment runs Workers in a single location, aggregating data from multiple regions, clusters, or datacenters. This approach works best for lower data volumes or when network peering already exists. Be aware that high-volume data transfers across regions or accounts may incur additional costs.
```
Suggest to tighten this up a bit for clarity but feel free to tweak as you see fit!

A hybrid model is a good compromise between the decentralized and centralized approaches, particularly for large wide-spread infrastructure deployments. For example, if you have six regions and in each region you have 10 Kubernetes clusters, rather than:

- Deploying Workers into each cluster, which results in 60 deployments
- Deploying Workers into one region and routing traffic across regions, which introduces a single point of failure

A hybrid approach is to have a dedicated Kubernetes cluster or manage instance group per region, which results in only six deployments. The 10 clusters in each region send data to the dedicated OPW deployment in that region.

## Optimize the instance

### Instance sizing

> Based on performance benchmarking for a pipeline that is using 12 processors to transform data, the Worker can handle approximately 1 TB per vCPU per day. For example, if you have 4 TB of events per day, you should provision enough compute plus headroom to account for your volumes. This could be three two-core machines or containers, or one six-core machine or container. Datadog recommends deploying Workers as part of an autoscaling group or deployed with [Horizontal Pod Autoscaling][7] enabled. Do not rely on a statically configured number of VMs or containers. This ensures that if the number of events spike, you can safely handle the traffic without data loss. It also ensures high availability should a Worker go down for any reason.

Reviewer: ```suggestion
Based on performance benchmarking for a pipeline that is using 12 processors to transform data, the Worker can handle approximately 1 TB per vCPU per day. For example, if you have 4 TB of events per day, you should provision enough compute plus headroom to account for your volumes. This could be three two-core machines or containers, or one six-core machine or container. Datadog recommends deploying Workers as part of an autoscaling group or deployed with [Horizontal Pod Autoscaling][7] enabled. Do not rely on a statically configured number of VMs or containers. This helps ensure you can safely handle traffic spikes without data loss and maintain high availability if a Worker goes down.
```
Just adjusting the last 2 sentences into 1 sentence for simplicity
